esp-hal/src/interrupt/mod.rs

```diff
@@ -135,19 +135,46 @@ pub trait InterruptConfigurable: crate::private::Sealed {
 pub struct InterruptHandler {
     f: extern "C" fn(),
     prio: Priority,
+    #[cfg(riscv)]
+    nested: bool,
```

Comment:
Hmm I'm not sure if I wrote this before or not, but if we stored the function's address, we don't need a new field.